specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-02-02-preview/managedClusters.json

```diff
@@ -5618,11 +5924,157 @@
+          "description": "The current managed cluster sku."
+        },
+        "enableRbac": {
+          "type": "boolean",
+          "description": "Whether the cluster has enabled Kubernetes Role-Based Access Control or not."
+        },
+        "networkProfile": {
+          "type": "object",
+          "$ref": "#/definitions/NetworkProfileForSnapshot",
+          "description": "The current network profile."
+        }
+      },
+      "description": "managed cluster properties for snapshot, these properties are read only."
+    },
+    "NetworkProfileForSnapshot": {
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "networkPlugin": {
+          "type": "string",
+          "enum": [
+            "azure",
+            "kubenet",
+            "none"
+          ],
+          "default": "kubenet",
+          "x-ms-enum": {
+            "name": "NetworkPlugin",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "azure",
+                "description": "Use the Azure CNI network plugin. See [Azure CNI (advanced) networking](https://docs.microsoft.com/azure/aks/concepts-network#azure-cni-advanced-networking) for more information."
+              },
+              {
+                "value": "kubenet",
+                "description": "Use the Kubenet network plugin. See [Kubenet (basic) networking](https://docs.microsoft.com/azure/aks/concepts-network#kubenet-basic-networking) for more information."
+              },
+              {
+                "value": "none",
+                "description": "Do not use a network plugin. A custom CNI will need to be installed after cluster creation for networking functionality."
+              }
+            ]
+          },
+          "description": "Network plugin used for building the Kubernetes network."
+        },
+        "networkPolicy": {
+          "type": "string",
+          "enum": [
+            "calico",
+            "azure"
+          ],
+          "x-ms-enum": {
+            "name": "NetworkPolicy",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "calico",
+                "description": "Use Calico network policies. See [differences between Azure and Calico policies](https://docs.microsoft.com/azure/aks/use-network-policies#differences-between-azure-and-calico-policies-and-their-capabilities) for more information."
+              },
+              {
+                "value": "azure",
+                "description": "Use Azure network policies. See [differences between Azure and Calico policies](https://docs.microsoft.com/azure/aks/use-network-policies#differences-between-azure-and-calico-policies-and-their-capabilities) for more information."
+              }
+            ]
+          },
+          "description": "Network policy used for building the Kubernetes network."
+        },
+        "networkMode": {
```

Comment:
is it possible to reference some predefined enum.

Reply:
Sure, made the change.